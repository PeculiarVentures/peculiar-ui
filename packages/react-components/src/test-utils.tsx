import React from 'react';
import {
  render,
  screen,
  fireEvent,
  waitFor,
  renderHook,
  act,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from './styles/theme_provider';

const renderWithWrapper = (
  ui: React.ReactElement,
) => (
  render(
    ui,
    {
      wrapper: ThemeProvider,
    },
  )
);

export {
  userEvent,
  fireEvent,
  renderWithWrapper,
  screen,
  waitFor,
  renderHook,
  act,
};
