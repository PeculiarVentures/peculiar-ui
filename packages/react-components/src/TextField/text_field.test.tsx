import React from 'react';
import { renderWithWrapper as render, screen, userEvent } from '../test-utils';
import { TextField } from './index';

describe('<TextField />', () => {
  describe('TextField render variants', () => {
    it('should render as default', () => {
      const { asFragment } = render(<TextField id="test-id" />);

      const input = screen.getByRole('textbox');

      expect(input).toBeInTheDocument();
      expect(input.tagName).toBe('INPUT');
      expect(input.getAttribute('type')).toBe('text');

      expect(asFragment()).toMatchSnapshot();
    });

    it('should have label', () => {
      const { asFragment } = render(
        <TextField
          label="Test label"
          id="test-id"
        />,
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it('should be disabled', () => {
      const { asFragment } = render(<TextField disabled id="test-id" />);

      expect(asFragment()).toMatchSnapshot();
    });

    it('should have required', () => {
      const { asFragment } = render(<TextField required id="test-id" />);

      expect(asFragment()).toMatchSnapshot();
    });

    it('should have name attr', () => {
      const { asFragment } = render(<TextField name="test-name" id="test-id" />);

      expect(asFragment()).toMatchSnapshot();
    });

    it('should have className', () => {
      const { asFragment } = render(
        <TextField
          className="div-cls"
          id="test-id"
          inputProps={{ className: 'input-cls' }}
        />,
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it('should forwards ref to div element', () => {
      const ref = React.createRef<HTMLDivElement>();

      render(<TextField ref={ref} />);

      const input = screen.getByRole('textbox');

      expect(input.closest('div')).toEqual(ref.current);
    });

    it('should forwards ref to input element', () => {
      const ref = React.createRef<HTMLInputElement>();

      render(<TextField inputRef={ref} />);

      const input = screen.getByRole('textbox');

      expect(input).toEqual(ref.current);
    });

    it('should have placeholder', () => {
      const { asFragment } = render(<TextField placeholder="test-placeholder" id="test-id" />);

      const input = screen.getByRole('textbox');

      expect(input.getAttribute('placeholder')).toBe('test-placeholder');

      expect(asFragment()).toMatchSnapshot();
    });

    it('should have defaultValue', () => {
      const { asFragment } = render(<TextField defaultValue="test-value" id="test-id" />);

      const input = screen.getByRole('textbox');

      expect(input.getAttribute('value')).toBe('test-value');

      expect(asFragment()).toMatchSnapshot();
    });

    it('should have value', () => {
      const onChange = jest.fn();

      const { asFragment } = render(
        <TextField
          value="test-value"
          onChange={onChange}
          id="test-id"
        />,
      );

      const input = screen.getByRole('textbox');

      expect(input.getAttribute('value')).toBe('test-value');

      expect(asFragment()).toMatchSnapshot();
    });

    it('should have error alert', () => {
      const { asFragment } = render(
        <TextField
          error
          errorText="Error message"
          id="test-id"
        />,
      );

      expect(screen.getByText('Error message')).toBeInTheDocument();

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('TextField render sizes', () => {
    const sizes: Array<React.ComponentProps<typeof TextField>['size']> = [
      'small',
      'medium',
      'large',
    ];

    sizes.forEach((size) => {
      it(`size "${size}"`, () => {
        const { asFragment } = render(
          <TextField size={size} id="test-id" />,
        );

        expect(asFragment()).toMatchSnapshot();
      });
    });
  });

  describe('TextField focus behaviour', () => {
    it('should have focus', async () => {
      render(<TextField />);

      const input = screen.getByRole('textbox');

      expect(document.activeElement).not.toEqual(input);
      await userEvent.tab();
      expect(document.activeElement).toEqual(input);
      expect(input).toHaveFocus();
    });

    it("shouldn't have focus when disabled has been passed to the component", async () => {
      render(<TextField disabled />);

      const input = screen.getByRole('textbox');

      expect(document.activeElement).not.toEqual(input);
      await userEvent.tab();
      expect(document.activeElement).not.toEqual(input);
      expect(input).not.toHaveFocus();
    });

    it('should have focus when autoFocus has been passed to the component', () => {
      render(<TextField autoFocus />);

      const input = screen.getByRole('textbox');

      expect(input).toHaveFocus();
    });

    it("shouldn't have focus when disabled & autoFocus has been passed to the component", async () => {
      render(<TextField disabled autoFocus />);

      const input = screen.getByRole('textbox');

      expect(input).not.toHaveFocus();
    });
  });

  describe('TextField keyboard behaviour', () => {
    it('should have correct value after user input', async () => {
      render(<TextField />);

      const input = screen.getByRole('textbox');

      await userEvent.type(input, 'test');
      expect(input).toHaveValue('test');
    });

    it("shouldn't change value after user input when readOnly has been passed to the component", async () => {
      render(<TextField readOnly defaultValue="test" />);

      const input = screen.getByRole('textbox');

      await userEvent.type(input, 'best');
      expect(input).toHaveValue('test');
    });
  });

  describe('TextField change behaviour', () => {
    it('should calls onChange after user input', async () => {
      const onChange = jest.fn();

      render(<TextField onChange={onChange} />);

      const input = screen.getByRole('textbox');

      expect(onChange).toBeCalledTimes(0);
      await userEvent.type(input, 'test');
      expect(onChange).toBeCalledTimes(4);
    });
  });
});
