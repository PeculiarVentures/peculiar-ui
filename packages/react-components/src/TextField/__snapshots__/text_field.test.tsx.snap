// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> TextField render sizes size "large" 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-black);
  font-weight: var(--pv-text-b3-weight);
  font-size: var(--pv-text-b3-size);
  line-height: var(--pv-text-b3-height);
  letter-spacing: var(--pv-text-b3-spacing);
  font-family: inherit;
  outline: none;
  box-sizing: border-box;
  width: 100%;
  border-radius: 4px;
  padding: 0 var(--pv-size-base-2);
  background-color: var(--pv-color-gray-1);
  border-style: solid;
  border-width: 1px;
  -webkit-transition: background-color 200ms,color 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,border-color 200ms;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  height: var(--pv-size-base-8);
  color: var(--pv-color-black);
  border-color: var(--pv-color-gray-8);
}

.emotion-0::-webkit-input-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0::-moz-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0:-ms-input-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0::placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0:hover {
  border-color: var(--pv-color-gray-10);
}

.emotion-0:disabled {
  cursor: not-allowed;
  border-color: var(--pv-color-gray-5);
  color: var(--pv-color-gray-7);
}

.emotion-0:not(:disabled)[aria-invalid] {
  border-color: var(--pv-color-wrong);
}

.emotion-0:not(:disabled):focus-visible {
  border-color: var(--pv-color-secondary);
}

<div>
    <input
      class="emotion-0"
      id="test-id"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;

exports[`<TextField /> TextField render sizes size "medium" 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-black);
  font-weight: var(--pv-text-b3-weight);
  font-size: var(--pv-text-b3-size);
  line-height: var(--pv-text-b3-height);
  letter-spacing: var(--pv-text-b3-spacing);
  font-family: inherit;
  outline: none;
  box-sizing: border-box;
  width: 100%;
  border-radius: 4px;
  padding: 0 var(--pv-size-base-2);
  background-color: var(--pv-color-gray-1);
  border-style: solid;
  border-width: 1px;
  -webkit-transition: background-color 200ms,color 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,border-color 200ms;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  height: var(--pv-size-base-7);
  color: var(--pv-color-black);
  border-color: var(--pv-color-gray-8);
}

.emotion-0::-webkit-input-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0::-moz-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0:-ms-input-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0::placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0:hover {
  border-color: var(--pv-color-gray-10);
}

.emotion-0:disabled {
  cursor: not-allowed;
  border-color: var(--pv-color-gray-5);
  color: var(--pv-color-gray-7);
}

.emotion-0:not(:disabled)[aria-invalid] {
  border-color: var(--pv-color-wrong);
}

.emotion-0:not(:disabled):focus-visible {
  border-color: var(--pv-color-secondary);
}

<div>
    <input
      class="emotion-0"
      id="test-id"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;

exports[`<TextField /> TextField render sizes size "small" 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-black);
  font-weight: var(--pv-text-c1-weight);
  font-size: var(--pv-text-c1-size);
  line-height: var(--pv-text-c1-height);
  letter-spacing: var(--pv-text-c1-spacing);
  font-family: inherit;
  outline: none;
  box-sizing: border-box;
  width: 100%;
  border-radius: 4px;
  padding: 0 var(--pv-size-base-2);
  background-color: var(--pv-color-gray-1);
  border-style: solid;
  border-width: 1px;
  -webkit-transition: background-color 200ms,color 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,border-color 200ms;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  height: var(--pv-size-base-6);
  color: var(--pv-color-black);
  border-color: var(--pv-color-gray-8);
}

.emotion-0::-webkit-input-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0::-moz-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0:-ms-input-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0::placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0:hover {
  border-color: var(--pv-color-gray-10);
}

.emotion-0:disabled {
  cursor: not-allowed;
  border-color: var(--pv-color-gray-5);
  color: var(--pv-color-gray-7);
}

.emotion-0:not(:disabled)[aria-invalid] {
  border-color: var(--pv-color-wrong);
}

.emotion-0:not(:disabled):focus-visible {
  border-color: var(--pv-color-secondary);
}

<div>
    <input
      class="emotion-0"
      id="test-id"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;

exports[`<TextField /> TextField render variants should be disabled 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-black);
  font-weight: var(--pv-text-b3-weight);
  font-size: var(--pv-text-b3-size);
  line-height: var(--pv-text-b3-height);
  letter-spacing: var(--pv-text-b3-spacing);
  font-family: inherit;
  outline: none;
  box-sizing: border-box;
  width: 100%;
  border-radius: 4px;
  padding: 0 var(--pv-size-base-2);
  background-color: var(--pv-color-gray-1);
  border-style: solid;
  border-width: 1px;
  -webkit-transition: background-color 200ms,color 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,border-color 200ms;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  height: var(--pv-size-base-7);
  color: var(--pv-color-black);
  border-color: var(--pv-color-gray-8);
}

.emotion-0::-webkit-input-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0::-moz-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0:-ms-input-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0::placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0:hover {
  border-color: var(--pv-color-gray-10);
}

.emotion-0:disabled {
  cursor: not-allowed;
  border-color: var(--pv-color-gray-5);
  color: var(--pv-color-gray-7);
}

.emotion-0:not(:disabled)[aria-invalid] {
  border-color: var(--pv-color-wrong);
}

.emotion-0:not(:disabled):focus-visible {
  border-color: var(--pv-color-secondary);
}

<div>
    <input
      class="emotion-0"
      disabled=""
      id="test-id"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;

exports[`<TextField /> TextField render variants should have className 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-black);
  font-weight: var(--pv-text-b3-weight);
  font-size: var(--pv-text-b3-size);
  line-height: var(--pv-text-b3-height);
  letter-spacing: var(--pv-text-b3-spacing);
  font-family: inherit;
  outline: none;
  box-sizing: border-box;
  width: 100%;
  border-radius: 4px;
  padding: 0 var(--pv-size-base-2);
  background-color: var(--pv-color-gray-1);
  border-style: solid;
  border-width: 1px;
  -webkit-transition: background-color 200ms,color 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,border-color 200ms;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  height: var(--pv-size-base-7);
  color: var(--pv-color-black);
  border-color: var(--pv-color-gray-8);
}

.emotion-0::-webkit-input-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0::-moz-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0:-ms-input-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0::placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0:hover {
  border-color: var(--pv-color-gray-10);
}

.emotion-0:disabled {
  cursor: not-allowed;
  border-color: var(--pv-color-gray-5);
  color: var(--pv-color-gray-7);
}

.emotion-0:not(:disabled)[aria-invalid] {
  border-color: var(--pv-color-wrong);
}

.emotion-0:not(:disabled):focus-visible {
  border-color: var(--pv-color-secondary);
}

<div
    class="div-cls"
  >
    <input
      class="input-cls emotion-0"
      id="test-id"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;

exports[`<TextField /> TextField render variants should have defaultValue 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-black);
  font-weight: var(--pv-text-b3-weight);
  font-size: var(--pv-text-b3-size);
  line-height: var(--pv-text-b3-height);
  letter-spacing: var(--pv-text-b3-spacing);
  font-family: inherit;
  outline: none;
  box-sizing: border-box;
  width: 100%;
  border-radius: 4px;
  padding: 0 var(--pv-size-base-2);
  background-color: var(--pv-color-gray-1);
  border-style: solid;
  border-width: 1px;
  -webkit-transition: background-color 200ms,color 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,border-color 200ms;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  height: var(--pv-size-base-7);
  color: var(--pv-color-black);
  border-color: var(--pv-color-gray-8);
}

.emotion-0::-webkit-input-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0::-moz-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0:-ms-input-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0::placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0:hover {
  border-color: var(--pv-color-gray-10);
}

.emotion-0:disabled {
  cursor: not-allowed;
  border-color: var(--pv-color-gray-5);
  color: var(--pv-color-gray-7);
}

.emotion-0:not(:disabled)[aria-invalid] {
  border-color: var(--pv-color-wrong);
}

.emotion-0:not(:disabled):focus-visible {
  border-color: var(--pv-color-secondary);
}

<div>
    <input
      class="emotion-0"
      id="test-id"
      type="text"
      value="test-value"
    />
  </div>
</DocumentFragment>
`;

exports[`<TextField /> TextField render variants should have error alert 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-black);
  font-weight: var(--pv-text-b3-weight);
  font-size: var(--pv-text-b3-size);
  line-height: var(--pv-text-b3-height);
  letter-spacing: var(--pv-text-b3-spacing);
  font-family: inherit;
  outline: none;
  box-sizing: border-box;
  width: 100%;
  border-radius: 4px;
  padding: 0 var(--pv-size-base-2);
  background-color: var(--pv-color-gray-1);
  border-style: solid;
  border-width: 1px;
  -webkit-transition: background-color 200ms,color 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,border-color 200ms;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  height: var(--pv-size-base-7);
  color: var(--pv-color-black);
  border-color: var(--pv-color-gray-8);
}

.emotion-0::-webkit-input-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0::-moz-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0:-ms-input-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0::placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0:hover {
  border-color: var(--pv-color-gray-10);
}

.emotion-0:disabled {
  cursor: not-allowed;
  border-color: var(--pv-color-gray-5);
  color: var(--pv-color-gray-7);
}

.emotion-0:not(:disabled)[aria-invalid] {
  border-color: var(--pv-color-wrong);
}

.emotion-0:not(:disabled):focus-visible {
  border-color: var(--pv-color-secondary);
}

.emotion-1 {
  margin: 0;
  color: var(--pv-color-wrong);
  font-weight: var(--pv-text-c2-weight);
  font-size: var(--pv-text-c2-size);
  line-height: var(--pv-text-c2-height);
  letter-spacing: var(--pv-text-c2-spacing);
  margin-top: 2px;
}

<div>
    <input
      aria-invalid="true"
      class="emotion-0"
      id="test-id"
      type="text"
      value=""
    />
    <p
      class="emotion-1"
    >
      Error message
    </p>
  </div>
</DocumentFragment>
`;

exports[`<TextField /> TextField render variants should have label 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin-bottom: 2px;
  display: inline-block;
}

.emotion-1 {
  margin: 0;
  color: var(--pv-color-gray-10);
  font-weight: var(--pv-text-c2-weight);
  font-size: var(--pv-text-c2-size);
  line-height: var(--pv-text-c2-height);
  letter-spacing: var(--pv-text-c2-spacing);
}

.emotion-2 {
  margin: 0;
  color: var(--pv-color-black);
  font-weight: var(--pv-text-b3-weight);
  font-size: var(--pv-text-b3-size);
  line-height: var(--pv-text-b3-height);
  letter-spacing: var(--pv-text-b3-spacing);
  font-family: inherit;
  outline: none;
  box-sizing: border-box;
  width: 100%;
  border-radius: 4px;
  padding: 0 var(--pv-size-base-2);
  background-color: var(--pv-color-gray-1);
  border-style: solid;
  border-width: 1px;
  -webkit-transition: background-color 200ms,color 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,border-color 200ms;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  height: var(--pv-size-base-7);
  color: var(--pv-color-black);
  border-color: var(--pv-color-gray-8);
}

.emotion-2::-webkit-input-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-2::-moz-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-2:-ms-input-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-2::placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-2:hover {
  border-color: var(--pv-color-gray-10);
}

.emotion-2:disabled {
  cursor: not-allowed;
  border-color: var(--pv-color-gray-5);
  color: var(--pv-color-gray-7);
}

.emotion-2:not(:disabled)[aria-invalid] {
  border-color: var(--pv-color-wrong);
}

.emotion-2:not(:disabled):focus-visible {
  border-color: var(--pv-color-secondary);
}

<div>
    <label
      class="emotion-0"
      for="test-id"
      id="test-id-label"
    >
      <span
        class="emotion-1"
      >
        Test label
      </span>
    </label>
    <input
      class="emotion-2"
      id="test-id"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;

exports[`<TextField /> TextField render variants should have name attr 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-black);
  font-weight: var(--pv-text-b3-weight);
  font-size: var(--pv-text-b3-size);
  line-height: var(--pv-text-b3-height);
  letter-spacing: var(--pv-text-b3-spacing);
  font-family: inherit;
  outline: none;
  box-sizing: border-box;
  width: 100%;
  border-radius: 4px;
  padding: 0 var(--pv-size-base-2);
  background-color: var(--pv-color-gray-1);
  border-style: solid;
  border-width: 1px;
  -webkit-transition: background-color 200ms,color 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,border-color 200ms;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  height: var(--pv-size-base-7);
  color: var(--pv-color-black);
  border-color: var(--pv-color-gray-8);
}

.emotion-0::-webkit-input-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0::-moz-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0:-ms-input-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0::placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0:hover {
  border-color: var(--pv-color-gray-10);
}

.emotion-0:disabled {
  cursor: not-allowed;
  border-color: var(--pv-color-gray-5);
  color: var(--pv-color-gray-7);
}

.emotion-0:not(:disabled)[aria-invalid] {
  border-color: var(--pv-color-wrong);
}

.emotion-0:not(:disabled):focus-visible {
  border-color: var(--pv-color-secondary);
}

<div>
    <input
      class="emotion-0"
      id="test-id"
      name="test-name"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;

exports[`<TextField /> TextField render variants should have placeholder 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-black);
  font-weight: var(--pv-text-b3-weight);
  font-size: var(--pv-text-b3-size);
  line-height: var(--pv-text-b3-height);
  letter-spacing: var(--pv-text-b3-spacing);
  font-family: inherit;
  outline: none;
  box-sizing: border-box;
  width: 100%;
  border-radius: 4px;
  padding: 0 var(--pv-size-base-2);
  background-color: var(--pv-color-gray-1);
  border-style: solid;
  border-width: 1px;
  -webkit-transition: background-color 200ms,color 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,border-color 200ms;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  height: var(--pv-size-base-7);
  color: var(--pv-color-black);
  border-color: var(--pv-color-gray-8);
}

.emotion-0::-webkit-input-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0::-moz-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0:-ms-input-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0::placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0:hover {
  border-color: var(--pv-color-gray-10);
}

.emotion-0:disabled {
  cursor: not-allowed;
  border-color: var(--pv-color-gray-5);
  color: var(--pv-color-gray-7);
}

.emotion-0:not(:disabled)[aria-invalid] {
  border-color: var(--pv-color-wrong);
}

.emotion-0:not(:disabled):focus-visible {
  border-color: var(--pv-color-secondary);
}

<div>
    <input
      class="emotion-0"
      id="test-id"
      placeholder="test-placeholder"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;

exports[`<TextField /> TextField render variants should have required 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-black);
  font-weight: var(--pv-text-b3-weight);
  font-size: var(--pv-text-b3-size);
  line-height: var(--pv-text-b3-height);
  letter-spacing: var(--pv-text-b3-spacing);
  font-family: inherit;
  outline: none;
  box-sizing: border-box;
  width: 100%;
  border-radius: 4px;
  padding: 0 var(--pv-size-base-2);
  background-color: var(--pv-color-gray-1);
  border-style: solid;
  border-width: 1px;
  -webkit-transition: background-color 200ms,color 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,border-color 200ms;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  height: var(--pv-size-base-7);
  color: var(--pv-color-black);
  border-color: var(--pv-color-gray-8);
}

.emotion-0::-webkit-input-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0::-moz-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0:-ms-input-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0::placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0:hover {
  border-color: var(--pv-color-gray-10);
}

.emotion-0:disabled {
  cursor: not-allowed;
  border-color: var(--pv-color-gray-5);
  color: var(--pv-color-gray-7);
}

.emotion-0:not(:disabled)[aria-invalid] {
  border-color: var(--pv-color-wrong);
}

.emotion-0:not(:disabled):focus-visible {
  border-color: var(--pv-color-secondary);
}

<div>
    <input
      class="emotion-0"
      id="test-id"
      required=""
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;

exports[`<TextField /> TextField render variants should have value 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-black);
  font-weight: var(--pv-text-b3-weight);
  font-size: var(--pv-text-b3-size);
  line-height: var(--pv-text-b3-height);
  letter-spacing: var(--pv-text-b3-spacing);
  font-family: inherit;
  outline: none;
  box-sizing: border-box;
  width: 100%;
  border-radius: 4px;
  padding: 0 var(--pv-size-base-2);
  background-color: var(--pv-color-gray-1);
  border-style: solid;
  border-width: 1px;
  -webkit-transition: background-color 200ms,color 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,border-color 200ms;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  height: var(--pv-size-base-7);
  color: var(--pv-color-black);
  border-color: var(--pv-color-gray-8);
}

.emotion-0::-webkit-input-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0::-moz-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0:-ms-input-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0::placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0:hover {
  border-color: var(--pv-color-gray-10);
}

.emotion-0:disabled {
  cursor: not-allowed;
  border-color: var(--pv-color-gray-5);
  color: var(--pv-color-gray-7);
}

.emotion-0:not(:disabled)[aria-invalid] {
  border-color: var(--pv-color-wrong);
}

.emotion-0:not(:disabled):focus-visible {
  border-color: var(--pv-color-secondary);
}

<div>
    <input
      class="emotion-0"
      id="test-id"
      type="text"
      value="test-value"
    />
  </div>
</DocumentFragment>
`;

exports[`<TextField /> TextField render variants should render as default 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-black);
  font-weight: var(--pv-text-b3-weight);
  font-size: var(--pv-text-b3-size);
  line-height: var(--pv-text-b3-height);
  letter-spacing: var(--pv-text-b3-spacing);
  font-family: inherit;
  outline: none;
  box-sizing: border-box;
  width: 100%;
  border-radius: 4px;
  padding: 0 var(--pv-size-base-2);
  background-color: var(--pv-color-gray-1);
  border-style: solid;
  border-width: 1px;
  -webkit-transition: background-color 200ms,color 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,border-color 200ms;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  height: var(--pv-size-base-7);
  color: var(--pv-color-black);
  border-color: var(--pv-color-gray-8);
}

.emotion-0::-webkit-input-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0::-moz-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0:-ms-input-placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0::placeholder {
  color: var(--pv-color-gray-9);
}

.emotion-0:hover {
  border-color: var(--pv-color-gray-10);
}

.emotion-0:disabled {
  cursor: not-allowed;
  border-color: var(--pv-color-gray-5);
  color: var(--pv-color-gray-7);
}

.emotion-0:not(:disabled)[aria-invalid] {
  border-color: var(--pv-color-wrong);
}

.emotion-0:not(:disabled):focus-visible {
  border-color: var(--pv-color-secondary);
}

<div>
    <input
      class="emotion-0"
      id="test-id"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;
